### PR TITLE
feat: add joyid support and replace metamask

### PR DESCRIPTION
**Pull request by SpectreMercury**
- [ ]  use JoyID Replace Metamask

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -94,7 +95,7 @@ function UIProvider({ children }: React.PropsWithChildren<{}>) {
 
 const config = {
   autoConnect: true,
-  connectors: [new MetaMaskConnector(), new JoyIdConnector()],
+  connectors: [new MetaMaskConnector, new JoyIdConnector()],
```
Comment (ahonn) [inline src/pages/_app.tsx:98]: Looks like the parentheses are missing

File: src/connectors/metamask.ts [hunk 2]
```diff
@@ -28,15 +28,16 @@ export default class MetaMaskConnector extends CKBConnector {
...
-    });
+    // const { publicClient, webSocketPublicClient } = configureChains(
+    //   [mainnet],
+    //   [publicProvider()],
+    // );
+    // this.config = createConfig({
+    //   autoConnect: true,
+    //   publicClient,
+    //   webSocketPublicClient,
+    // });
```
Comment (ahonn) [inline src/connectors/metamask.ts:39]: I don’t quite understand why this is done. 
If you want to disable MetaMask, just remove `MetaMaskConnector` in `connectors: [new MetaMaskConnector, new JoyIdConnector()]`

File: src/connectors/joyId.ts [hunk 3]
```diff
@@ -48,7 +43,7 @@ export default class JoyIdConnector extends CKBConnector {
       return;
     }
     const ethAddress = await connect();
```
Comment (ahonn) [inline src/connectors/joyId.ts:45]: After using `@joyid/ckb` to connect, should be rename `ethAddress` to `authData`.
see: https://docs.joyid.dev/guide/ckb/connect#get-started

**Comment by ahonn** [comment]
I suggest not to remove MetaMask support first.
So as not to prevent users operate the spores of mint in the past.

File: src/pages/_app.tsx (hunk 1 above)
Comment (SpectreMercury) [reply src/pages/_app.tsx:98]: remove this due to the hotel network not stable, will fix

File: src/connectors/metamask.ts (hunk 2 above)
Comment (SpectreMercury) [reply src/connectors/metamask.ts:39]: remove this due to the hotel network not stable, will fix

**Comment by SpectreMercury** [comment]
> I suggest not to remove MetaMask support first. So as not to prevent users operate the spores of mint in the past.

The previous concealment was mainly due to compatibility issues arising from the OmniLock update, which temporarily suspended support for MetaMask. Support has now been restored.

**Review by ashuralyk — APPROVED** [review]

**Comment by ashuralyk** [comment]
I cannot find some problems and just not to block the work.

**Pull request merged**